src/pytest_plugins/filler/pre_alloc.py

```diff
@@ -298,18 +301,66 @@ def sha256_from_string(s: str) -> int:
     return int.from_bytes(sha256(s.encode("utf-8")).digest(), "big")
 
 
+ALL_FIXTURE_FORMAT_NAMES: List[str] = []
+
+for spec in BaseTest.spec_types.values():
+    for labeled_fixture_format in spec.supported_fixture_formats:
+        name = (
+            labeled_fixture_format.label
+            if isinstance(labeled_fixture_format, LabeledFixtureFormat)
+            else labeled_fixture_format.format_name.lower()
+        )
+        if name not in ALL_FIXTURE_FORMAT_NAMES:
+            ALL_FIXTURE_FORMAT_NAMES.append(name)
+
+# Sort by length, from longest to shortest, since some fixture format names contain others
+# so we are always sure to catch the longest one first.
+ALL_FIXTURE_FORMAT_NAMES.sort(key=len, reverse=True)
+
+
+@pytest.fixture(scope="function")
+def node_id_for_entropy(request: pytest.FixtureRequest, fork: Fork | None) -> str:
+    """
+    Return the node id with the fixture format name and fork name stripped.
+
+    Used in cases where we are filling for pre-alloc groups, and we take the name of the
+    test as source of entropy to get a deterministic address when generating the pre-alloc
+    grouping.
+
+    Removing the fixture format and the fork name from the node id before hashing results in the
+    contracts and senders addresses being the same across fixture types and forks for the same
+    test.
+    """
+    node_id: str = request.node.nodeid
+    if fork is None:
+        # FIXME: Static tests don't have a fork, so we need to get it from the node.
```

Comment:
```suggestion
        # FIXME: Static tests don't have a fork, so we need to get it from the node.
```
Should we create an issue?

Reply:
I'm not sure if it's solvable tbh, this is a quirk of how we hack pytest to do static tests, it's not priority IMO but we can keep it in mind.